internal/state/graph/httproute.go

```diff
@@ -524,12 +524,12 @@ func validatePathMatch(
 	}
 
 	if *path.Type != v1beta1.PathMatchPathPrefix {
-		valErr := field.NotSupported(fieldPath, *path.Type, []string{string(v1beta1.PathMatchPathPrefix)})
+		valErr := field.NotSupported(fieldPath.Child("type"), *path.Type, []string{string(v1beta1.PathMatchPathPrefix)})
```

Comment:
Is there a test we can add for this fix?

Reply:
we happen to have an affected test (which I forgot to update https://github.com/nginxinc/nginx-kubernetes-gateway/actions/runs/4767096667/jobs/8474949006?pr=587 ), but in general the code that test our validation don't check error messages though

Reply:
That's right. Since these error messages are part of the API, I wonder if it makes sense to test them.

Reply:
The corresponding conditions - ex. Type=Accepted Status=False Reason=Unsupported value - are for sure part of the API for sure -- but the  messages - I don't think so. So we _can't_ change Condition,Status,Reason (it's gonna be a breaking change), but we _can_ change the messages itself - ex. improve their format, include more details. etc.  I would not make the message a part of the API.

At the same time, since there is a logic in generating those messages -- generating the affected field path - we can add tests for that. That will test that logic and also make sure that the right error was return (in the code path we expected).

However, not sure if it is that beneficial. Message-related bugs are minor imho.

Reply:
> Message-related bugs are minor imho.

Good point, I agree. It would be nice to have tests that verify the field path, but I don't think it should be a release or PR blocker.